### ISSUE TITLE
Convert documentation from Sphinx to MkDocs

### ISSUE THREAD

**Issue by tleonhardt** [python-cmd2/cmd2#1396]
One idea I've been toying around with in my head is porting cmd2's documentation from using Sphinx to using [MkDocs](https://www.mkdocs.org/).

ReadTheDocs supports MkDocs and MkDocs has been rapidly growing in popularity. In summary it is similar to Sphinx but uses Markdown instead of reStructuredText.

I figure that in this day and age, essentially all developers know Markdown, but knowledge of reStructuredText is much less common. So it would likely help others help us maintain the documentation.

As long as nobody objects, I'll likely start experimenting with this soon. But I'd like to hear what other contributors think ...

**Comment by tleonhardt** [comment]
I did some research and looked at seven popular Python projects which use `MkDocs` and they all use the [Material Theme for MkDocs](https://squidfunk.github.io/mkdocs-material/).

For reference, here are the projects I looked at:
- [AutoGPT](https://github.com/Significant-Gravitas/AutoGPT)
- [hello-algo](https://github.com/krahets/hello-algo)
- [FastAPI](https://github.com/fastapi/fastapi)
- [ruff](https://github.com/astral-sh/ruff)
- [uv](https://github.com/astral-sh/uv)
- [polars](https://github.com/pola-rs/polars)
- [pydantic](https://github.com/pydantic/pydantic)

**Comment by tleonhardt** [comment]
The [MkDocs Catalog](https://github.com/mkdocs/catalog) has ranked lists of MkDocs themes, markdown extensions, and plugins.

**Comment by kotfu** [comment]
Markdown is great, and I'm all in favor of switching our documentation to it. I have no preference whether we stick with Sphinx or switch to MkDocs. No preference on theme either, Material is commonly used and I think works just fine.

**Comment by kotfu** [comment]
Should we decide to make this change, I'm happy to help convert/migrate files from RST to Markdown.

**Comment by tleonhardt** [comment]
ReadTheDocs has this [demo repository](https://github.com/readthedocs/test-builds/tree/mkdocs-material) that shows a correct bare-bones configuration for using `Material for MkDocs` with `ReadTheDocs`.

**Comment by tleonhardt** [comment]
@kotfu @kmvanbrunt I'm now very close to done with the WIP PR: #1397 

It is now fully functional - docs have been converted to Markdown, MkDocs Material theme build is working, looks good, and is integrated with our GitHub Action CI and all Sphinx stuff has been removed.

The only thing remaining is for me to make some tweaks so it integrates nicely with `ReadTheDocs`. If one or both of you would like to start taking a look, I'd appreciate any feedback.

TL;DR on using the new stuff:

Start a live HTML server that auto-reloads on changed Markdown files:

```sh
mkdocs serve
```
or equivalently:
```sh
inv livehtml
```
Build the docs:

```sh
mkdocs build
```
or equivalently:

```sh
inv docs
```

**Comment by tleonhardt** [comment]
PR is now done and no longer a WIP

**Comment by tleonhardt** [comment]
This change is now merged into all active branches and the documentation has been fully converted to use **Markdown** with the [Material theme for MkDocs](https://squidfunk.github.io/mkdocs-material/).

The build documentation is viewable as the [ReadTheDocs latest](https://cmd2.readthedocs.io/en/latest/).
